src/bsoncxx/builder/core.hpp

```diff
@@ -263,6 +268,11 @@ class BSONCXX_API core {
     void append(const oid& value);
 
     ///
+    /// Append a decimal128 object as a BSON Decimal128.
+    ///
+    void append(const decimal128& value);
```

Comment:
For whatever reason *all* the appends are const ref.  Even b_int64, which is literally just an in64_t.